### PR TITLE
[Bug #8915] - Dragging over/onto an iFrame no longer works

### PR DESCRIPTION
**Pull request by qooalt**
Fix for the Bug #8915 This commit fixes in my case the odd behaviour of "blocking" the dragover and drop events when crossing an iframe while dragging elements. See links below to the playground.

3.5.1 (ok)
http://tinyurl.com/og6l97e

5.0 (wrong)
http://tinyurl.com/nw3ur95

See:  https://github.com/qooxdoo/qooxdoo/issues/8915

### CODE REVIEW

File: framework/source/class/qx/event/handler/DragDrop.js [hunk 1]
```diff
@@ -575,6 +576,7 @@ qx.Class.define("qx.event.handler.DragDrop",
 
         this.__manager.addListener(this.__root, "pointermove", this._onPointermove, this);
         this.__manager.addListener(this.__root, "pointerup", this._onPointerup, this, true);
+        this.__manager.addListener(this.__root, "track", this._onPointermove, this);
       }
     },
```
Comment (oetiker) [inline framework/source/class/qx/event/handler/DragDrop.js]: please elaborate on adding "track"

File: framework/source/class/qx/event/handler/DragDrop.js [hunk 2]
```diff
@@ -665,6 +669,10 @@ qx.Class.define("qx.event.handler.DragDrop",
      */
     _getDelta : function(e)
     {
+      if (e.getDelta) {
+        return e.getDelta();
+      }
+      
       if (!this.__startConfig) {
```
Comment (oetiker) [inline framework/source/class/qx/event/handler/DragDrop.js]: who supports getDelta ? this also seems to be unrelated to the iframe fix

File: framework/source/class/qx/event/handler/DragDrop.js [hunk 3]
```diff
@@ -613,7 +615,9 @@ qx.Class.define("qx.event.handler.DragDrop",
       }
 
       // find current hovered droppable
-      var el = e.getTarget();
+      // TODO: Review this (before the fix it was "e.getTarget()", but I feel this to be better)
+      var doc = this.__manager.getWindow().document;
+      var el = doc.elementFromPoint(e.getDocumentLeft(), e.getDocumentTop());
       var cursor = this.getCursor();
```
Comment (oetiker) [inline framework/source/class/qx/event/handler/DragDrop.js]: does it only not work when the iframe content is coming from another domain ... that would explain why your solution works as it does not look at the content of the iframe

**Comment by oetiker** [comment]
ping ...

**Comment by level420** [comment]
@qooalt Please have a look into this PR again. There are some open questions.

**Comment by level420** [comment]
As @qooalt does not react: I think we should close the PR

**Comment by oetiker** [comment]
is the bug not reproducible ?

**Comment by level420** [comment]
I don't know, have currently no use case for that.

**Comment by level420** [comment]
closing due to inactivity. feel free to reopen. The bugreport wil remain open.

**Comment by level420** [comment]
@oetiker why did you re-open this PR?

**Comment by oetiker** [comment]
I re-opened it because we ran into this but in osparc

**Comment by level420** [comment]
@oetiker does this PR solve that bug? the corresponding bug is here https://github.com/qooxdoo/qooxdoo/issues/8915 and still open

File: framework/source/class/qx/event/handler/DragDrop.js (hunk 1 above)
Comment (level420) [reply framework/source/class/qx/event/handler/DragDrop.js]: Its http://www.qooxdoo.org/devel/api/#qx.event.type.Track where track is derived from http://www.qooxdoo.org/devel/api/#qx.event.type.Pointer only adding one additional method ```getDelta``` which is documented as:
Returns a map with the calculated delta coordinates and axis, relative to the position on trackstart event.

File: framework/source/class/qx/event/handler/DragDrop.js (hunk 2 above)
Comment (level420) [reply framework/source/class/qx/event/handler/DragDrop.js]: this is supporte by the http://www.qooxdoo.org/devel/api/#qx.event.type.Track event, which was registerd above.

**Comment by oetiker** [comment]
it seems that @qooalt is a throwaway account created only to generate this PR ... hence no response.

**Comment by hkollmann** [comment]
Done with https://github.com/qooxdoo/qooxdoo/pull/9838
